target/linux/ath79/image/generic.mk

```diff
@@ -138,6 +138,21 @@ define Device/dlink_dir-825-b1
 endef
 TARGET_DEVICES += dlink_dir-825-b1
 
+define Device/dlink_dir-859-a1
+$(Device/seama)
+  ATH_SOC := qca9563
+  DEVICE_TITLE := D-LINK DIR-859 A1
+  IMAGE_SIZE := 15872k
+  DEVICE_PACKAGES :=  kmod-usb-core kmod-usb2 kmod-ath10k-ct ath10k-firmware-qca988x-ct
+  SEAMA_SIGNATURE := wrgac37_dlink.2013gui_dir859
+  IMAGE/factory.bin := \
```

Comment:
If you can't use the factory.bin recipe from `Device/seama`, drop the `Device/seama` factory.bin recipe. No need to carry around unused code, which might (or not) be required at some time. It should be added if required.

Reply:
The factory.bin must be used for the first time upgrading from the factory firmware.
Then upgrade the sysupgrade.bin from OpenWrt to avoid triggering u-boot's firmware checksum validation.

Reply:
Sorry, misunderstood your words.
Normally we expect the jffs2 end-of-filesystem marker is not covered by the seama header.
Because the seama header has a md5 checksum field. If the jffs2 marker is covered by the header, once OpenWrt booted up the jffs2 marker will be erased and the checksum will become incorrect.
However the d-link's firmware and u-boot require the whole firmware to be covered by the header. The factory.bin will only pass the u-boot's validation for the first time it boots up. The u-boot will refuse to boot the firmware after reboot and enter the failsafe mode. So the sysupgrade.bin must be upgraded in the factory.bin at the first boot.

So, I assume this is a special case for d-link's firmware, and choose to override the recipe instead of replacing the Device/seama.